interp/src/flatten/primitives/prim_trait.rs

```diff
@@ -158,10 +160,13 @@ pub trait RaceDetectionPrimitive: Primitive {
     /// Get a reference to the underlying primitive. Unfortunately cannot add an
     /// optional default implementation due to size rules
     fn as_primitive(&self) -> &dyn Primitive;
+
+    fn clone_boxed(&self) -> Box<dyn RaceDetectionPrimitive>;
```

Comment:
Might be worth giving this a different name to avoid aliasing with the
`clone_boxed` on `Primitive`. Maybe `clone_boxed_rd`?